src/contract_spec.ts

```diff
@@ -605,21 +675,15 @@ export class ContractSpec {
         `failed to find entry ${name} in union {udt.name().toString()}`
       );
     }
-    let res: Union<any> = { tag: name, values: undefined };
+    let res: Union<any> = { tag: name };
     if (
       entry.switch().value ===
       xdr.ScSpecUdtUnionCaseV0Kind.scSpecUdtUnionCaseTupleV0().value
     ) {
       let tuple = entry.value() as xdr.ScSpecUdtUnionCaseTupleV0;
       let ty = tuple.type();
-      //@ts-ignore
-      let values = [];
-      for (let i = 0; i < ty.length; i++) {
-        let v = this.scValToNative(vec[i + 1], ty[i]);
-        values.push(v);
-      }
-      let r = { tag: name, values };
-      return r;
+      let values = ty.map((entry, i) => this.scValToNative(vec![i + 1], entry));
```

Comment:
wait what the heck is `vec!` in js???

Reply:
lol it is annotating that the vec exists. But I had the same thought.